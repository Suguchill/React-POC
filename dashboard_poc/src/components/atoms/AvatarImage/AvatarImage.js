import React from "react";
import PropTypes from "prop-types";

const AvatarImage = (props) => {
  const {
    src,
    className,
    alt,
    ...attrs
  } = props;

  return (
    <img
      src={src}
      className={className}
      alt={alt}
      { ...attrs } 
    />
  );
}

AvatarImage.propTypes = {
  src: PropTypes.string,
  className: PropTypes.string,
  alt: PropTypes.string,
};

AvatarImage.defaultProps = {
    className: 'rounded-circle',
    alt: 'Image',
    src: ''
}

export default AvatarImage;