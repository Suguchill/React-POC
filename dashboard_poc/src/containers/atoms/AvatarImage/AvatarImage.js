import React from "react";
import PropTypes from "prop-types";

const AvatarImage = (props) => {
  const {
    src,
    className,
    alt,
    ...attrs
  } = props;

  return (
    <img
      src={src}
      className={className}
      alt={alt}
      { ...attrs } 
    />
  );
}

AvatarImage.propTypes = {
  src: PropTypes.string,
  className: PropTypes.string,
  alt: PropTypes.string,
};

AvatarImage.defaultProps = {
  alt: 'Image',
}

export default AvatarImage;