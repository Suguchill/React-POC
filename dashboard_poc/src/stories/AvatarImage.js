import React from 'react';
import PropTypes from 'prop-types';
import './avatarImage.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import Home from './assets/Home.png'

export const AvatarImage = ({src, className, alt, ...attrs}) => {
  return (
    <img
      src={src}
      className={className}
      alt={alt}
      { ...attrs } 
    />
  );
};

AvatarImage.propTypes = {
  className: PropTypes.string,
  src: PropTypes.string,
  alt: PropTypes.string
};

AvatarImage.defaultProps = {
    className: 'rounded-circle',
    src: Home,
    alt: 'Image'
};