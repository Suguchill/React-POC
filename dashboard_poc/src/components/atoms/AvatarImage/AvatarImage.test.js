import React from 'react';
import { cleanup, render } from "@testing-library/react"
import AvatarImage from './AvatarImage'
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});

afterEach(cleanup)

describe("Button Component", () => {
    const AvatarImageProps = {
        id: "avatarimg",
        className: "img-fluid",
        alt: "Avatar Image",
        src: ""
    }

    it("Rendering content",() => {
        const {getByAltText} = render(<AvatarImage {...AvatarImageProps} />)
        const value = getByAltText("Avatar Image")
        expect(value.getAttribute("id")).toBe("avatarimg")
        expect(value.getAttribute("class")).toBe("img-fluid")
        // expect(value.getAttribute("src")).toBe(ProfileImg)
    })

    it('Render correctly', () => {
        const component = shallow(<AvatarImage debug />);
        expect(component).toMatchSnapshot();
    });
})