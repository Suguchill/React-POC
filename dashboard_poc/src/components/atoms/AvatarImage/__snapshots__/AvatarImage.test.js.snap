// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarImage Component Render correctly 1`] = `ShallowWrapper {}`;
